plugins/Live/API.php

```diff
@@ -289,6 +295,28 @@ public function getMostRecentVisitorId($idSite, $segment = false)
         return $visitor->getVisitorId();
     }
 
+    /**
+     * Returns the very first visit for the given visitorId
+     *
+     * @internal
+     *
+     * @param $idSite
+     * @param $visitorId
+     *
+     * @return DataTable
+     */
+    public function getFirstVisitForVisitorId($idSite, $visitorId)
+    {
+        Piwik::checkUserHasViewAccess($idSite);
+
+        $model = new Model();
+        $data = $model->queryLogVisits($idSite, false, false, false, 0, 1, $visitorId, false, 'ASC');
```

Comment:
Here we should it seems check that `$visitorId` is set, and if not fail? as otherwise we may return possibly some other visits in `queryLogVisits`?

Reply:
It now returns an empty datatable if the visitorid is empty